airbyte-protocol/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -74,45 +74,60 @@ definitions:
     properties:
       state_type:
         "$ref": "#/definitions/AirbyteStateType"
+      stream:
+        "$ref": "#/definitions/AirbyteStreamState"
+      global:
+        "$ref": "#/definitions/AirbyteGlobalState"
       data:
         description: "(Deprecated) the state data"
         type: object
         existingJavaType: com.fasterxml.jackson.databind.JsonNode
-      global:
-        "$ref": "#/definitions/AirbyteStateBlob"
-      streams:
-        type: array
-        items:
-          "$ref": "#/definitions/AirbyteStreamState"
 
   AirbyteStateType:
     type: string
     description: >
       The type of state the other fields represent.
       If not set, the state data is interpreted as GLOBAL and should be read from the `data` field for backwards compatibility.
-      GLOBAL means that the state should be read from `global` and means that it represents the state for all the streams.
-      PER_STREAM means that the state should be read from `streams`. Each item in the list represents the state for the associated stream.
+      GLOBAL means that the state should be read from `global` and means that it represents the state for all the streams. It contains one shared
+      state and individual stream states.
+      PER_STREAM means that the state should be read from `stream`. The state present in this field correspond to the isolated state of the
+      associated stream description.
     enum:
       - GLOBAL
       - PER_STREAM
-
   AirbyteStreamState:
     type: object
-    description: "per stream state data"
     additionalProperties: false
     required:
-      - name
+      - streamDescriptor
       - state
     properties:
-      name:
-        description: "Stream name"
-        type: string
+      streamDescriptor:
+        "$ref": "#/definitions/StreamDescriptor"
       state:
         "$ref": "#/definitions/AirbyteStateBlob"
+  AirbyteGlobalState:
+    type: object
+    additionalProperties: false
+    required:
+      - sharedState
+    properties:
+      sharedState:
+        "$ref": "#/definitions/AirbyteStateBlob"
+      streamStates:
+        type: array
+        items:
+          "$ref": "#/definitions/AirbyteStreamState"
+  StreamDescriptor:
+    type: object
+    additionalProperties: false
```

Comment:
is there any reason not to set this to `true`? i dunno if we'll ever change how we identify a stream, but i'm not sure there's any value to locking ourselves in.

Reply:
I wanted to avoid having the object to generate the extra field additionalProperties where it is not needed. I set it to true for future compatibility.